{
	"schema_version": "1.6.0",
	"id": "GHSA-m884-279h-32v2",
	"modified": "2023-12-06T01:00:36.068760Z",
	"published": "2020-03-30T20:09:31Z",
	"aliases": [
		"BIT-symfony-2020-5274",
		"CVE-2020-5274"
	],
	"summary": "Exceptions displayed in non-debug configurations in Symfony",
	"details": "Description\n-----------\n\nWhen `ErrorHandler` renders an exception HTML page, it uses un-escaped properties from the related Exception class to render the stacktrace. The security issue comes from the fact that the stacktraces were also displayed in non-`debug` environments.\n\nResolution\n----------\n\nThe `ErrorHandler` class now escapes all properties coming from the related Exception, and the stacktrace is not displayed anymore in non-`debug` environments.\n\nThe patches for this issue are available [here](https://github.com/symfony/symfony/commit/cf80224589ac05402d4f72f5ddf80900ec94d5ad) and [here](https://github.com/symfony/symfony/commit/629d21b800a15dc649fb0ae9ed7cd9211e7e45db) for branch 4.4.\n\nCredits\n-------\n\nI would like to thank Luka Sikic for reporting & Yonel Ceruto and Jérémy Derussé for fixing the issue.",
	"severity": [
		{
			"type": "CVSS_V3",
			"score": "CVSS:3.1/AV:N/AC:L/PR:L/UI:R/S:U/C:L/I:L/A:N"
		}
	],
	"affected": [
		{
			"package": {
				"ecosystem": "Packagist",
				"name": "symfony/http-foundation",
				"purl": "pkg:composer/symfony/http-foundation"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "4.4.0"
						},
						{
							"fixed": "4.4.4"
						}
					]
				}
			],
			"versions": [
				"v4.4.0",
				"v4.4.1",
				"v4.4.2",
				"v4.4.3"
			],
			"database_specific": {
				"source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2020/03/GHSA-m884-279h-32v2/GHSA-m884-279h-32v2.json"
			}
		},
		{
			"package": {
				"ecosystem": "Packagist",
				"name": "symfony/http-foundation",
				"purl": "pkg:composer/symfony/http-foundation"
			},
			"ranges": [
				{
					"type": "ECOSYSTEM",
					"events": [
						{
							"introduced": "5.0.0"
						},
						{
							"fixed": "5.0.4"
						}
					]
				}
			],
			"versions": [
				"v5.0.0",
				"v5.0.1",
				"v5.0.2",
				"v5.0.3"
			],
			"database_specific": {
				"source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2020/03/GHSA-m884-279h-32v2/GHSA-m884-279h-32v2.json"
			}
		}
	],
	"references": [
		{
			"type": "WEB",
			"url": "https://github.com/symfony/symfony/security/advisories/GHSA-m884-279h-32v2"
		},
		{
			"type": "ADVISORY",
			"url": "https://nvd.nist.gov/vuln/detail/CVE-2020-5274"
		},
		{
			"type": "WEB",
			"url": "https://github.com/symfony/symfony/commit/629d21b800a15dc649fb0ae9ed7cd9211e7e45db"
		},
		{
			"type": "WEB",
			"url": "https://github.com/symfony/symfony/commit/cf80224589ac05402d4f72f5ddf80900ec94d5ad"
		}
	],
	"database_specific": {
		"cwe_ids": [
			"CWE-209"
		],
		"github_reviewed": true,
		"github_reviewed_at": "2020-03-30T19:36:11Z",
		"nvd_published_at": null,
		"severity": "MODERATE"
	}
}
